tools/build/Makefile.vars

```diff
@@ -286,7 +286,9 @@ COMMON_CFLAGS += -g -fno-lto
 endif
 
 ifeq (y,$(BUILD_TYPE_RELEASE))
-COMMON_CFLAGS += $(RELEASE_COMMON_CFLAGS)
+ifeq (,$(shell echo $(COMMON_CFLAGS) | grep -e "-O0"))
+COMMON_CFLAGS += $(FORTIFY_CFLAGS)
+endif
```

Comment:
Well, I've tried building with
```bash
BUILD_TYPE_RELEASE=n make
```
and it seems to ignore that, whereas building with
```bash
COMMON_CFLAGS="-g -O0"
```
correctly prepends ```-g -O0``` to the COMMON_CFLAGS.

The bottom line is that COMMON_CFLAGS may be set from outside the build environment, so the build environment has to account for this. Now, most of the time the COMMON_CFLAGS arriving from the outside are orthogonal to what we're adding inside the build environment, but in this case the existing COMMON_CFLAGS need to take precedence.